### PR TITLE
Initial vulkan implementation

### PR DESCRIPTION
**Pull request by Beyley**
Contains the absolute most barebones vulkan backend (only instance creation)

Everything else is either `unreachable` or `@panic("TODO")`

warning: ***very*** early  and ugly code

`src/vulkan/impl.zig` and `src/dusk_impl.zig` are the points of interest here, rest are taken from `mach-gpu` for the struct definitions (but may have extra Ziggy wrappers left over from the copy that still need to be removed)

Only compiles on Linux, vulkan-loader needs to be worked out for the other platforms

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,13 @@ jobs:
         uses: actions/checkout@v2
       - name: Setup Zig
         run: |
-          sudo apt install xz-utils
+          sudo apt install xz-utils
```
Comment (alichraghi) [inline .github/workflows/ci.yml:13]: blank space

File: src/dawn.zig
Comment (alichraghi) [inline src/dawn.zig:1]: no dawn. get rid of this

**Review by alichraghi — COMMENTED** [review]
is there anything stopping us from using zig package manager to import types from `hexops/mach-gpu`?
i believe we only need `dusk_impl.zig` and `vulkan/`?

File: src/dawn.zig
Comment (Beyley) [reply src/dawn.zig:1]: > no dawn. get rid of this

This file contains extra structs for extensions provided by dawn from what i can see, eventually maybe a `wgpu.zig` would be a good idea so we can expose the wgpu extensions too

**Comment by Beyley** [comment]
> is there anything stopping us from using zig package manager to import types from `hexops/mach-gpu`? i believe we only need `dusk_impl.zig` and `vulkan/`?

I dont think anything prevents us from doing so atm, but i copied them incase i need to modify them

**Pull request merged**